src/univers/versions.py

```diff
@@ -343,3 +344,86 @@ def __gt__(self, other):
         if not isinstance(other, self.__class__):
             return NotImplemented
         return gentoo.vercmp(self.value, other.value) > 0
+
+
+@attr.s(frozen=True, order=False, eq=False, hash=True)
+class LegacyOpensslVersion(Version):
+    @classmethod
+    def is_valid(cls, string):
+        return OpensslVersion.is_valid_legacy(string)
+
+    def __eq__(self, other):
+        if isinstance(other, SemverVersion):
+            return False
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        return self.value == other.value
+
+    def __lt__(self, other):
+        """
+        All legacy scheme is behind those using semver.
+        """
+        if isinstance(other, SemverVersion):
+            return True
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        re_vers1 = re.findall(r"^(\d+\.\d+\.\d+)(.*)$", self.value)[0]
+        re_vers2 = re.findall(r"^(\d+\.\d+\.\d+)(.*)$", other.value)[0]
+        if int(re_vers1[0].replace(".", "")) < int(re_vers2[0].replace(".", "")):
+            return True
+        return re_vers1[0] == re_vers2[0] and re_vers1[1] < re_vers2[1]
+
+    def __gt__(self, other):
+        """
+        All openssl version using semver is ahead of legacy scheme.
+        """
+        if isinstance(other, SemverVersion):
+            return False
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        re_vers1 = re.findall(r"^(\d+\.\d+\.\d+)([a-z-]+)$", self.value)[0]
+        re_vers2 = re.findall(r"^(\d+\.\d+\.\d+)([a-z-]+)$", other.value)[0]
+        if int(re_vers1[0].replace(".", "")) > int(re_vers2[0].replace(".", "")):
+            return True
+        return re_vers1[0] == re_vers2[0] and re_vers1[1] > re_vers2[1]
+
+    def __le__(self, other):
+        return self.__lt__(other) or self.__eq__(other)
+
+    def __ge__(self, other):
+        return self.__gt__(other) or self.__eq__(other)
+
+
+@attr.s(frozen=True, order=False, eq=False, hash=True)
+class OpensslVersion:
+    """
+    Handles the object creation for openssl depending on
+    whether version is legacy or semver.
+    """
```

Comment:
```suggestion
    """
    Openssl versions are of two types
        - Legacy versions: Implemented in OpensslLegaryVersion
        - New versions: Semver
    """
```

Reply:
Also, have they explicitly named these version types as ``Legacy`` and ``New`` ?

Reply:
No, they don't call it new or Legacy.. they're just calling it old and contemporary format..🤷